include/packetgraph/thread.h

```diff
@@ -0,0 +1,70 @@
+/* Copyright 2017 Outscale SAS
+ *
+ * This file is part of Butterfly.
+ *
+ * Butterfly is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as published
+ * by the Free Software Foundation.
+ *
+ * Butterfly is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Butterfly.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef _PG_THREAD_H
+#define _PG_THREAD_H
+
+/* TODO: error handeling ? */
+
+#include <packetgraph/errors.h>
+#include <packetgraph/graph.h>
+
+enum pg_thread_state {
+	PG_CANT_RECIVE_STATE = -1,
+	PG_THREAD_STOPPED,
+	PG_THREAD_BROKEN,
+	PG_THREAD_RUNNING
+};
+
+#define pg_thread_states_to_str(state)		\
+	(pg_thread_str_states[state + 1])
```

Comment:
why not [stringify](https://gcc.gnu.org/onlinedocs/gcc-5.3.0/cpp/Stringification.html) ?